File: frontend/src/app/shared/service/admin/inscriptionref/ConfirmationTokenAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {ConfirmationTokenDto} from 'src/app/shared/model/inscriptionref/ConfirmationToken.model';
import {ConfirmationTokenCriteria} from 'src/app/shared/criteria/inscriptionref/ConfirmationTokenCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ConfirmationTokenAdminService extends AbstractService<ConfirmationTokenDto, ConfirmationTokenCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): ConfirmationTokenDto {
        return new ConfirmationTokenDto();
    }

    public constrcutCriteria(): ConfirmationTokenCriteria {
        return new ConfirmationTokenCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/confirmationToken/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/inscriptionref/ConfirmationTokenAdminService.java
package ma.zs.alc.service.facade.admin.inscriptionref;

import java.util.List;
import ma.zs.alc.bean.core.inscriptionref.ConfirmationToken;
import ma.zs.alc.dao.criteria.core.inscriptionref.ConfirmationTokenCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface ConfirmationTokenAdminService extends  IService<ConfirmationToken,ConfirmationTokenCriteria>  {

    List<ConfirmationToken> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantId(Long id);




}
